p2p/simulations/events_test.go

```diff
@@ -0,0 +1,166 @@
+package simulations
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+)
+
+func init() {
+	log.Root().SetHandler(log.LvlFilterHandler(log.LvlInfo, log.StreamHandler(os.Stderr, log.TerminalFormat(true))))
+}
+
+// This test confirms (and demonstrates) using simulation events
```

Comment:
This comment doesn't make sense to me. What is this test testing?

Reply:
As discussed on the phone; it's not really a test as much as it is boilerplate code to demonstrate how to wait for cruical network events instead of just sleeping or polling.

The code is implemented in the `swarm/network/simulations/discovery/dynamic_test.go` test, so this code is actually redundant; we could just use that other implementation as the reference (if we want to use this method).

Anyway, we should converge on _one_ method to check if the network is "ready," even "ready" to be checked for health.